scripts/release/audit_rosdistro.py

```diff
@@ -0,0 +1,135 @@
+
+    print('## Packages failing on all platforms\n Release should be rolled back:')
+    for mp in sorted(missing_packages['all']):
+        print(' - %s ' % mp)
+
+    def get_package_repository_link(dist_file, pkg_name):
+        """Return the best guess of the url for filing a ticket against the package."""
+        pkg = dist_file.release_packages[pkg_name]
+        repo_name = pkg.repository_name
+        repo = dist_file.repositories[repo_name]
+        if repo.source_repository and repo.source_repository.url:
+            return repo.source_repository.url
+        if repo.release_repository and repo.release_repository.url:
+            return repo.release_repository.url
+        return None
+
+    for bf_name in build_files.keys():
+        print('## Packages failing for buildfile %s\n\nPackages should be blacklisted:' % bf_name)
+        # TODO(tfoote) use rosdistro API to print the release build config for editing
+        recommended_blacklists = sorted(missing_packages[bf_name]['all'] - missing_packages['all'])
+        recommended_actions += len(recommended_blacklists)
+        for rb in recommended_blacklists:
+            print(' - %s:' % rb)
+            jenkins_urls = get_jenkins_job_urls(
+                rosdistro_name,
+                config.jenkins_url,
+                bf_name,
+                build_files[bf_name].get_targets_list())
+            url = get_package_repository_link(dist_file, rb)
+            print('   - Suggested ticket location [%s](%s)' % (url, url))
+            print('')
+            print('   Title:')
+            print('')
+            print('       %s in %s fails to build on %s targets' % (rb, rosdistro_name, bf_name))
+            print('')
+            print('   Body:')
+            print('')
+            print('       The package %s in %s has been detected as not building'
+                  % (rb, rosdistro_name) +
+                  ' on all platforms in the buildfile %s.' % (bf_name) +
+                  ' The release manager for %s will consider disabling' % (rosdistro_name) +
+                  ' this build if it continues to fail to build.')
+            print('       - jenkins_urls:')
+            for target, ju in jenkins_urls.items():
+                target_str = ' '.join([x for x in target])
+                url = ju.format(pkg=rb)
+                print('          - [%s](%s)' % (target_str, url))
+                # TODO(tfoote) embed build status when buildfarm has https
+                # print('    - %s [![Build Status](%s)](%s)' % (' '.join([x for x in target]),
+                #       ju.format(pkg = rb) + '/badge/icon', ju.format(pkg = rb)))
+            print('       This is being filed because this package is about to be blacklisted.'
+                  ' If this ticket is resolved please review whether it can be removed from'
+                  ' the blacklist that should cross reference here.')
+            print('')
```

Comment:
Consider moving this padding space up one level or not printing "Packages should be blacklisted:" unless there are some. The Dashing result has several stanzas that appear like this:
```markdown
## Packages failing for buildfile default

Packages should be blacklisted:
## Packages failing for buildfile ubv8

```
which doesn't affect the rendered markdown but made looking at the plaintext harder to follow.

Reply:
I improved the wording in 39089a8cfff8e9650eb2f3ce4373b06589750a69